pkg/zrouter/zmiddlewares/topNRequests.go

```diff
@@ -0,0 +1,103 @@
+package zmiddlewares
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/zondax/golem/pkg/logger"
+	"github.com/zondax/golem/pkg/zcache"
+	"github.com/zondax/golem/pkg/zrouter/auth"
+	"net/http"
+	"strings"
+	"time"
+)
+
+const (
+	tokenDetailsTTLDefault = 45 * time.Minute
+	PathUsageByJWTKey      = "jwt_path_usage"
+	defaultTTL             = time.Hour
+)
+
+func TopRequestTokensMiddleware(zCache zcache.RemoteCache, tokenDetailsTTL, usageMetricTTL time.Duration) func(next http.Handler) http.Handler {
```

Comment:
Can we rename this to something related to JWT? We may add the same thing for top N fil addresses requests, and the logic is pretty much the same, but different extract logic.